Change OAuth2.0 Exchange authentication to Basic Auth

riksu-raksu opened the pull request:
We haven't found a way to configure OAuth on the new cloud O365 without violating security policies that the cloud caretaker has in place.

All OAuth configurations that would technically work with a server-to-server integration like Respa has require too wide permissions to the Respa credentials. This is why we move to use Basic Auth instead.

Microsoft was going to disable Basic Auth in the cloud, but COVID got them postpone that plan. Their plan is to still disable Basic Auth, but a bit later, sometime during 2021.

So this change puts us on a borrowed time and we need a different long-term plan. However this gets us alive and kicking in production for now.

The documentation I added perhaps does not belong to a code repository but rather in a project documentation. However I don't know where that would be, I added it here so that it's at least written down somewhere and does not get lost to time.

jopesy commented:
IMHO the reversion of commit ef4b9d4 and changing to basic auth should be separate commits. But it's alright like this if we need to get this into production quickly. Approved.

Commit:
Change Oauth2.0 Exchange authentication to NTLM

We haven't found a way to configure OAuth on the new cloud O365 without
violating security policies that the cloud caretaker has in place.

All OAuth configurations that would technically work with a server-to-server
integration like Respa has require too wide permissions to the Respa
credentials. This is why we go back to NTLM and rethink our approach.

This reverts commit ef4b9d4f3183f706adc73ff218a59eb95fd46f08.

Commit:
Change NTLM Exchange authentication to Basic Auth

NTLM authentication is not available in the new Office 365 cloud, so we
need to replace it.

Microsoft was going to disable Basic Auth in the cloud, but COVID got
them postpone that plan. Their plan is to still disable Basic Auth,
but a bit later, sometime during 2021.

So this change puts us on a borrowed time and we need a different
long-term plan. However this gets us alive in kicking in production
for now.

riksu-raksu commented:
@jopesy No problem changing that (done and committed), but can you tell me why is the pipeline failing. It's about coverage I see, does something need to be done about it? `Hits` is -23, do we need to get that number up?

jopesy commented:
@riksu-raksu That should be OK.
